### PR TITLE
ARROW-4692: [Flight] Explain sidecar in a bit more detail

### PR DESCRIPTION
**Pull request by emkornfield**

### CODE REVIEW

File: format/Flight.proto [hunk 1]
```diff
@@ -287,7 +287,9 @@ message FlightData {
 
   /*
    * The actual batch of Arrow data. Preferably handled with minimal-copies
-   * coming last in the definition to help with sidecar patterns.
+   * coming last in the definition to help with sidecar patterns (it is
+   * expected that some implementations will parse this field off the wire
```
Comment (pitrou) [inline format/Flight.proto]: I would replace "parse" with "fetch". Part of the intent is _not_ have to parse at all.
Of course, the way we do it currently is not exactly idiomatic.

Comment (emkornfield) [reply format/Flight.proto]: Done.

**Review by pitrou — APPROVED** [review]
+1 from me

**Comment by pitrou** [comment]
Rebased.

**Comment by pitrou** [comment]
Thank you @emkornfield !

**Pull request closed without merging**